### PR TITLE
Update TimeStore to return timestamp-value pairs

### PR DESCRIPTION
**Pull request by afein**
cc @vishh 
Updated TimeStore.Get method to return structs of timestamp and value

### CODE REVIEW

File: store/time_store.go [hunk 1]
```diff
@@ -28,8 +28,13 @@ type TimeStore interface {
 	// If 'start' is zero, it returns all the elements up until 'end'.
 	// If 'end' is zero, it returns all the elements from 'start'.
 	// If both 'start' and 'end' are zero, it returns all the elements in the cache.
-	Get(start, end time.Time) []interface{}
+	Get(start, end time.Time) []TimePoint
```
Comment (vishh) [inline store/time_store.go]: Update Put API to take in TimePoint.

File: store/in_memory_test.go [hunk 2]
```diff
@@ -33,9 +33,9 @@ func TestNilInsert(t *testing.T) {
 	assert.Error(t, store.Put(time.Now(), nil))
 }
 
-func expectElements(t *testing.T, data []interface{}) {
+func expectElements(t *testing.T, data []TimePoint) {
```
Comment (vishh) [inline store/in_memory_test.go]: Validate the timestamps as well.

**Comment by vishh** [comment]
Just a couple of comments. LGTM otherwise

**Comment by afein** [comment]
@vishh @rjnagal ready to merge

**Comment by vishh** [comment]
LGTM. Thanks for the PR @afein

**Pull request merged**